### PR TITLE
Rule S2068: filter string literal that contains the wordlist item

### LINKED ISSUE #2868: Rule S2068: filter string literal that contains the wordlist item
[RSPEC-2068](https://jira.sonarsource.com/browse/RSPEC-2068)

The implementation of this rule highly rely on symbol names matching wordlist items to raise issues. The downside of this is that it raises many FPs when constants are used to avoid duplicated stings:

```csharp
string password = "Password"; // Compliant
string pwd = "pwd"; // Compliant

public class S2068 {

  private const string PASSWORD = "Password"; // Compliant
  private const string PASSWORD_INPUT = "[id='password']"; // Compliant
  private const string PASSWORD_PROPERTY = "custom.password"; // Compliant
  private const string TRUSTSTORE_PASSWORD = "trustStorePassword"; // Compliant
  private const string CONNECTION_PASSWORD = "connection.password"; // Compliant
  private const string RESET_PASSWORD = "/users/resetUserPassword"; // Compliant

// ...
}
```

In most string constants use cases the wordlist item is present in both the symbol name and the string value.
The new approach is to avoid raising issues when **the wordlist item is present in both symbol name and literal string value**.
Exception made for the following use cases that are still True Positives:

```csharp
string params = "user=admin&password=Password123"; // Sensitive
string sqlserver = "pgsql:host=localhost port=5432 dbname=test user=postgres password=postgres"; // Sensitive
```

### PR DESCRIPTION
**Pull request by pavel-mikula-sonarsource**
Fixes #2868

### CODE REVIEW

File: sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Rules/Hotspots/DoNotHardcodeCredentialsBase.cs [hunk 1]
```diff
@@ -151,11 +151,17 @@ private IEnumerable<string> FindCredentialWords(string variableName, string vari
                     return Enumerable.Empty<string>();
                 }
 
-                var credentialWordsFound = variableName
+                var credentialWordsFound = (variableName ==null || variableName.Any(x => char.IsLower(x)) ? variableName : variableName.ToLower()) // Prepare "PASSWORD" for SplitCamelCaseToWords()
```
Comment (christophe-zurn-sonarsource) [inline sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Rules/Hotspots/DoNotHardcodeCredentialsBase.cs]: I think this fix deserves a separate ticket: fixing camelcase split for fully uppercase words should be directly done inside the `SplitCamelCaseToWords` method IMO.
But since it would impact other part of the project, I suggest to do it in a separate commit.
WDYT ?

**Comment by pavel-mikula-sonarsource** [comment]
SplitCamelCaseToWords is addressed in PR #3188 that needs to be merged first.

**Comment by pavel-mikula-sonarsource** [comment]
Rebasing issue was fixed. There was duplicate commit adding duplicate expected issue as in #3188  PR.
